lib/collections/addon/instance-initializers/route-styles.ts

```diff
@@ -0,0 +1,39 @@
+import Controller from '@ember/controller';
+import EngineInstance from '@ember/engine/instance';
+import podNames from 'ember-component-css/pod-names';
+
+export function initialize(engineInstance: EngineInstance) {
+    const Router = engineInstance.lookup('router:main');
+
+    Router.reopen({
+        didTransition(infos: any[]) {
+            this._super(infos);
+
+            const classes = new Set<string>();
+
+            for (const info of infos) {
+                // TODO: Generically fix the pod-names for engines
```

Comment:
Yeah, we should fork/fix/upstream this, or at least give a good bug report. ember-component-css does support engines, so they'll want to know how it breaks for routes.